numpy/core/setup.py

```diff
@@ -902,6 +902,7 @@ def generate_umath_c(ext, build_dir):
             join('src', 'umath', 'ufunc_object.c'),
             join('src', 'umath', 'scalarmath.c.src'),
             join('src', 'umath', 'ufunc_type_resolution.c'),
+            join('src', 'umath', 'logical_gufuncs.c.src'),
```

Comment:
Am playing around with this as an example `gufunc` that might be passed through `__array_ufunc__` in #8247, and looking at this, I think you should also add `logical_gufuncs.h` to the `umath_deps` list just below here.